jackson/src/test/java/com/baeldung/jackson/deserialization/JacksonMapDeserializeTest.java

```diff
@@ -0,0 +1,67 @@
+package com.baeldung.jackson.deserialization;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.baeldung.jackson.entities.MyPair;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class JacksonMapDeserializeTest {
+
+	private Map<MyPair, String> map;
+	private Map<MyPair, MyPair> cmap;
+
+	@Test
+	public void whenSimpleMapDeserialize_thenCorrect()
+			throws JsonParseException, JsonMappingException, IOException {
+
+		final String jsonInput = "{\"key\": \"value\"}";
+		final ObjectMapper mapper = new ObjectMapper();
+		TypeReference<HashMap<String, String>> typeRef = new TypeReference<HashMap<String, String>>() {
+		};
+
+		final Map<String, String> map = mapper.readValue(jsonInput, typeRef);
+		final String expectedOutput = "{key=value}";
+		Assert.assertEquals(expectedOutput, map.toString());
```

Comment:
A more appropriate test for these deserializations might be to assert that, given the key, you get the expected value from the map.
For example:
Assert.assertEquals("value", map.get("key"));